src/signals/incident-management/containers/IncidentDetail/components/AddNote/index.js

```diff
@@ -1,109 +1,103 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { FormBuilder, FieldGroup, Validators } from 'react-reactive-form';
+import styled from 'styled-components';
+import { Button, themeSpacing } from '@datapunt/asc-ui';
 
-import FieldControlWrapper from 'signals/incident-management/components/FieldControlWrapper';
-import TextAreaInput from 'signals/incident-management/components/TextAreaInput';
+import TextArea from 'components/TextArea';
+import Label from 'components/Label';
 import { PATCH_TYPE_NOTES } from 'models/incident/constants';
 
-import './style.scss';
+const NewNoteButton = styled(Button)`
+  margin: ${themeSpacing(2, 2, 2, 0)};
+`;
 
-class AddNote extends React.Component { // eslint-disable-line react/prefer-stateless-function
-  form = FormBuilder.group({ // eslint-disable-line react/sort-comp
-    text: [null, Validators.required],
-  });
+const SaveNoteButton = styled(Button)`
+  margin-right: ${themeSpacing(2)};
+`;
 
-  constructor(props) {
-    super(props);
+const AddNote = ({ id, onPatchIncident }) => {
+  const areaRef = useRef(null);
+  const [showForm, setShowForm] = useState(false);
+  const [note, setNote] = useState('');
 
-    this.state = {
-      formVisible: false,
-    };
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault();
 
-    this.showForm = this.showForm.bind(this);
-    this.hideForm = this.hideForm.bind(this);
-  }
+      const notes = [{ text: note }];
 
-  handleSubmit = event => {
-    event.preventDefault();
-    const notes = [{ text: this.form.value.text }];
-    this.props.onPatchIncident({
-      id: this.props.id,
-      type: PATCH_TYPE_NOTES,
-      patch: { notes },
-    });
-
-    this.form.reset();
-    this.hideForm();
-  }
+      onPatchIncident({
+        id,
+        type: PATCH_TYPE_NOTES,
+        patch: { notes },
+      });
 
-  showForm() {
-    this.form.reset();
-    this.setState({ formVisible: true });
-  }
+      setShowForm(false);
```

Comment:
Is here not deeded to clean note state? Otherwise when you click on new, you'll get the old note text.

Reply:
Wow, nice catch. I completely missed that!

Reply:
Fixed